packages/url-utils/src/utils/build-early-exit-match.ts

```diff
@@ -16,14 +21,14 @@ function escapeRegExp(string) {
  * @param {boolean} [options.ignoreProtocol=true] - Whether to strip protocol from URLs
  * @returns {string|null} Regex pattern matching any configured base URL, or null if none configured
  */
-function buildEarlyExitMatch(siteUrl, options = {}) {
+function buildEarlyExitMatch(siteUrl: string, options: BuildEarlyExitMatchOptions = {}): string | null {
     const candidates = [siteUrl, options.imageBaseUrl, options.filesBaseUrl, options.mediaBaseUrl]
-        .filter(Boolean)
-        .map((value) => {
+        .filter((value): value is string => Boolean(value))
```

Comment:
`value is string` doesn't match the actual implementation 🤔 

The `Boolean(value)` check is making sure it's a non empty string - I think we should change this line to the following which is slightly more correct and at least enforces the string type

```ts
.filter((value: string | null): value is string => typeof value === 'string' && value.length > 0)
```